grr/server/grr_response_server/gui/ui/lib/api/api_interfaces.ts

```diff
@@ -10,22 +10,115 @@
 
 export declare interface AnyObject {
   '@type'?: string;
-  [key: string]: undefined|null|string|number|boolean|object;
+  [key: string]: undefined | null | string | number | boolean | object;
+}
+
+/** Api RdfValueFieldDescriptor proto mapping. */
+export declare interface RdfValueFieldDescriptor {
+  readonly name?: string;
+  readonly doc?: string;
+  readonly friendlyName?: string;
+}
+
+/** Api RdfValueDescriptor proto mapping. */
+export declare interface RdfValueDescriptor {
+  readonly name?: string;
+  readonly doc?: string;
+  readonly fields?: ReadonlyArray<RdfValueFieldDescriptor>;
+}
+
+/** ApiAllRdfValues proto mapping. */
+export declare interface ApiAllRdfValues {
+  readonly items?: ReadonlyArray<RdfValueDescriptor>;
 }
 
 /**
  * int64, fixed64, and DecimalString in Protobuf are serialized as string in
  * JSON because JS loses precision for big numeric types. During
  * deserialization, both decimal strings and numbers are accepted.
  */
-export type DecimalString = string|number;
+export type DecimalString = string | number;
+
+/**
+ * ApiUser protomapping.
+ */
+export declare interface ApiUser {
+  readonly username?: string;
+  readonly lastLogon?: string;
+  readonly fullName?: string;
+  readonly homedir?: string;
+  readonly uid?: number;
+  readonly gid?: number;
+  readonly shell?: string;
+}
+
+/**
+ * Network address proto mapping
+ */
+export declare interface ApiNetworkAddress {
+  readonly addressType?: string;
+  readonly packedBytes?: string;
+}
+
+/**
+ * Network Interface proto mapping
+ */
+export declare interface ApiInterface {
+  readonly macAddress?: string;
+  readonly ifname?: string;
+  readonly addresses?: ReadonlyArray<ApiNetworkAddress>;
+}
 
 /**
  * KnowledgeBase proto mapping.
  */
 export declare interface ApiKnowledgeBase {
   readonly fqdn?: string;
+  readonly timeZone?: string;
+
   readonly os?: string;
+  readonly osMajorVersion?: number;
+  readonly osMinorVersion?: number;
+}
+
+/**
+ * ApiClientInformation proto mapping.
+ */
+export declare interface ApiClientInformation {
+  readonly clientName?: string;
+  readonly clientVersion?: number;
+  readonly revision?: number;
+  readonly buildTime?: string;
+  readonly clientBinaryName?: string;
+  readonly clientDescription?: string;
+  readonly labels?: string[];
```

Comment:
Please use immutable arrays in the type definitions (also below): `ReadonlyArray<string>`.